Mathlib/Algebra/Category/ModuleCat/Presheaf.lean

```diff
@@ -497,4 +497,79 @@ def unitHomEquiv (M : PresheafOfModules R) :
     ext X
     exact (LinearMap.ringLmapEquivSelf (R.obj X) ℤ (M.obj X)).apply_symm_apply (s.val X)
 
+section module_over_initial
+
+/-!
+## `PresheafOfModules R ⥤ Cᵒᵖ ⥤ ModuleCat (R.obj X)` when `X` is initial
+
+When `X` is initial, we have `Module (R.obj X) (M.obj c)` for any `c : Cᵒᵖ`.
+
+-/
+
+/--
+Implementation of the functor `PresheafOfModules R ⥤ Cᵒᵖ ⥤ ModuleCat (R.obj X)`
+when `X` is initial.
+
+The functor is implemented as, on object level `M ↦ (c ↦ M(c))` where the `R(X)`-module structure
+on `M(c)` is given by restriction of scalars along the unique morphism `R(c) ⟶ R(X)`; and on
+morphism level `(f : M ⟶ N) ↦ (c ↦ f(c))`.
+-/
+@[simps]
+noncomputable def forgetToPresheafModuleCatObj
+    (X : Cᵒᵖ) (hX : Limits.IsInitial X) (M : PresheafOfModules.{v} R) :
+    Cᵒᵖ ⥤ ModuleCat (R.1.obj X) where
+  obj c :=
+    ModuleCat.restrictScalars (R.1.map (hX.to c)) |>.obj <| M.obj c
+  map := fun {c₁ c₂} f =>
+    { toFun := fun x => M.presheaf.map f x
+      map_add' := M.presheaf.map f |>.map_add
+      map_smul' := fun r (m : M.presheaf.obj c₁) => by
+        change M.presheaf.map f (R.1.map (hX.to c₁) _ • m) = R.1.map (hX.to c₂) _ • _
```

Comment:
Are we missing (d)simp lemmas to get the goal in the right shape? I would prefer if we could avoid `change`.

Reply:
This is resolved by explicitly tell Lean `m : ModuleCat.restrictScalars ....`